pandas/core/series.py

```diff
@@ -2331,11 +2330,7 @@ def align(self, other, join='outer', axis=None, level=None, copy=True,
 
     @Appender(generic._shared_docs['rename'] % _shared_doc_kwargs)
     def rename(self, index=None, **kwargs):
-        is_scalar_or_list = (
-            (not com.is_sequence(index) and not callable(index)) or
-            (com.is_list_like(index) and not isinstance(index, MutableMapping))
-        )
-        if is_scalar_or_list:
```

Comment:
`com.is_list_like(dict)` is True, and  I need to treat dicts differently than lists.

But if we're going to be raising on `Series.rename(list)` anway, I won't have to worry about lists at all...